collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -69,13 +69,21 @@
 
     private static final Pattern snakeCasePattern = Pattern.compile("([a-z0-9])([A-Z])");
 
+    private final JmxMBeanPropertyCache jmxMBeanPropertyCache;
```

Comment:
Wouldn't this make more sense in the Scraper?

Reply:
I'm happy to change that, but my read of the code was that, Scraper is initialized in `public List<MetricFamilySamples> collect()`. Since that function is public, `collect()` might be invoked multiples times; so sharing the cache would provide most benefit there. I'm happy to move to scraper if you think that is best.

Reply:
Right you are.